### DISCUSSION TITLE
A small feedback

### DISCUSSION THREAD

**Discussion by vorcigernix** [WalletConnect/web-examples#83] [Ideas]
Hi, 
I am trying to build a very similar thing to the react wallet v2. It is way more complicated than it should be. There are things I need to understand from a code logic perspective and that's fine. But I also battle some state management that I have never seen before with UI library that definitely cannot be described as widely used. A nitpick: with a SPA-like logic using modals where most people use routing and "pages" - which is super weird given your use of next.js. 
I am sure there could be good reasons for all of those. What I am saying is that the sample code should be as simple as possible, not as well architected as possible.
Thanks for building this anyway, the code is actually way better place to understand how it works than a documentation. I appreciate your effort. If I succeed building some dumbed-down version I can share it with you in case there is more people like me.

**Comment by RasKemal** [comment]
Hi, Adam I'm currently building a web3 wallet and I agree with you about the code. I was wondering have you managed to build a simple version of this wallet?

**Comment by vorcigernix** [comment]
Well, sort of. Price for that is really horrible code full of hacks at
https://github.com/vorcigernix/burney
I actually abandoned the idea.
